File: 2025/2025-02-24/fizzbuzz.py
def fizzbuzz(n: int):
    results = []
    for i in range(1, n):
        match (i % 3, i % 5):
            case (0, 0):
                results.append("FizzBuzz")
            case (0, _):
                results.append("Fizz")
            case (_, 0):
                results.append("Buzz")
            case _:
                results.append(str(i))d
    return results

File: 2025/2025-02-10/sample_prj_2025_02_10/lib/sample_prj_2025_02_10/fizzbuzz.rb
# frozen_string_literal: true

#
# FizzBuzzを実行するためのクラス
#
class SamplePrj20250210::FizzBuzz
  #
  # コンストラクタ
  #
  # @param [Range] range FizzBuzzを表示する範囲
  #
  def initialize(range)
    @range = range
  end

  def run
    @range.map do |n|
      fizzbuzz n
    end
  end

  def fizzbuzz(target)
    if (target % 15).zero?
      'fizzbuzz'
    elsif (target % 3).zero?
      'fizz'
    elsif (target % 5).zero?
      'buzz'
    else
      target.to_s
    end
  end
end
